Resources/public/js/views/services/ez-universaldiscoveryviewservice.js

```diff
@@ -22,6 +22,51 @@ YUI.add('ez-universaldiscoveryviewservice', function (Y) {
      */
     Y.eZ.UniversalDiscoveryViewService = Y.Base.create('universalDiscoveryViewService', Y.eZ.ViewService, [Y.eZ.SideViewService], {
         /**
+         *
+         * @method _load
+         * @protected
+         * @param {Function} callback
+         */
+        _load: function (callback) {
+            var startingLocation = new Y.eZ.Location(),
+                path = [];
+
+            if (this.get('parameters').startingLocationId) {
+                startingLocation.set('id', this.get('parameters').startingLocationId);
+                this._loadStartingLocationPath(startingLocation, Y.bind(function(locationPath) {
+                    startingLocation.get('path').unshift(this.get('virtualRootLocation'));
+                    this.get('parameters').startingLocation = startingLocation;
+                    callback();
+                }, this));
+            } else {
+                this.get('parameters').startingLocation = this.get('virtualRootLocation');
+                callback();
+            }
+        },
+
+        /**
+         * `Get the path of the UDW starting Location.
+         *
+         * @method _loadStartingLocationPath
+         * @protected
+         * @param {eZ.Location} startingLocation
+         * @param {Function} callback
+         */
+        _loadStartingLocationPath: function (startingLocation, callback) {
+            var options = {api: this.get('capi')};
+
+            startingLocation.load(options, function (error) {
+                if (!error) {
```

Comment:
something has to be done when an error occurs otherwise the user will be stuck in front of the app loader. In case of error while loading the startingLocation or its path, we should probably act as if the UDW was launched without a startingLocationId.